src/mbgl/map/map.cpp

```diff
@@ -727,44 +625,30 @@ void Map::updateAnnotation(AnnotationID id, const Annotation& annotation) {
 
 void Map::removeAnnotation(AnnotationID annotation) {
     impl->annotationManager.removeAnnotation(annotation);
-    impl->onUpdate(Update::AnnotationStyle | Update::AnnotationData);
+    impl->onUpdate(Update::AnnotationData);
 }
 
 #pragma mark - Feature query api
 
 std::vector<Feature> Map::queryRenderedFeatures(const ScreenCoordinate& point, const RenderedQueryOptions& options) {
```

Comment:
We can, but it also facilitates `AnnotationIDs queryPointAnnotations(const ScreenBox&);`. Moving logic specific to Annotations into the renderer is maybe not that clean. It does clean up the different `RendererFrontend` implementations though..